### PR TITLE
Changed Newton iteration to log-log iteration in CompositeViscoPlastic

### PR DESCRIPTION
**Pull request by bobmyhill**
This PR changes the CompositeViscoPlastic rheology in ASPECT to use a log-stress Newton scheme. This requires fewer iterations to converge and should be generally more robust.

This PR builds on #5977.

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).
* [x] I have tested my new feature locally to ensure it is correct.
* [x] I have modified a testcase for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: include/aspect/material_model/rheology/composite_visco_plastic.h [hunk 1]
```diff
@@ -93,24 +94,19 @@ namespace aspect
...
            */
           std::pair<double, double>
-          compute_strain_rate_and_derivative (const double creep_stress,
-                                              const double pressure,
-                                              const double temperature,
-                                              const DiffusionCreepParameters diffusion_creep_parameters,
-                                              const DislocationCreepParameters dislocation_creep_parameters,
-                                              const PeierlsCreepParameters peierls_creep_parameters,
-                                              const DruckerPragerParameters drucker_prager_parameters) const;
+          calculate_log_strain_rate_and_deriv(const std::array<std::pair<double, double>, 5> &logarithmic_strain_rates_and_stress_derivatives) const;
```
Comment (tjhei) [inline include/aspect/material_model/rheology/composite_visco_plastic.h:109]:
```suggestion
          calculate_log_strain_rate_and_derivative(const std::array<std::pair<double, double>, 5> &logarithmic_strain_rates_and_stress_derivatives) const;
```

File: source/material_model/rheology/composite_visco_plastic.cc [hunk 2]
```diff
@@ -418,23 +426,27 @@ namespace aspect
...
-            AssertThrow(prm.get_bool("Use plastic damper") && prm.get_double("Plastic damper viscosity") > 0.,
-                        ExcMessage("If Drucker-Prager plasticity is included in the rheological formulation, you must use a viscous damper with a positive viscosity."));
+            active_flow_mechanisms.push_back(3);
           }
 
-        AssertThrow(use_diffusion_creep == true || use_dislocation_creep == true || use_peierls_creep == true || use_drucker_prager == true,
-                    ExcMessage("You need to include at least one deformation mechanism."));
+        active_flow_mechanisms.push_back(4);
 
+        AssertThrow(active_flow_mechanisms.size() != 1,
```
Comment (tjhei) [inline source/material_model/rheology/composite_visco_plastic.cc:448]:
```suggestion
        AssertThrow(active_flow_mechanisms.size() > 1,
```

File: source/material_model/rheology/drucker_prager_power.cc [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+        // circumscribing (Glerum et al 2018), middle circumscribing or
+        // inscribing the Mohr-Coulomb yield surface.
+        // See for instance Owen & Hinton, Finite Elements in Plasticity, 1980.
+        // Here the middle circumscribing approach is taken.
+        const double stress_inv_part = 1. / (std::sqrt(3.0) * (3.0 + sin_phi));
+
+        // Initial yield stress (no stabilization terms)
+        const double yield_stress = ( (dim==3)
+                                      ?
+                                      ( 6.0 * cohesion * cos_phi + 6.0 * pressure * sin_phi) * stress_inv_part
```
Comment (tjhei) [inline source/material_model/rheology/drucker_prager_power.cc:91]: can you please put () around the if and the else term?

**Review by tjhei — APPROVED** [review]

File: source/material_model/rheology/drucker_prager_power.cc (hunk 3 above)
Comment (bobmyhill) [reply source/material_model/rheology/drucker_prager_power.cc]: This was implemented in #5976, so I've made the changes there and rebased #5977 and this PR.

**Comment by bobmyhill** [comment]
Hi @tjhei, thanks for your review. I realised that there was no test for when some flow laws were switched off. I've now added a new test.

**Comment by tjhei** [comment]
Can you update test results?

**Comment by bobmyhill** [comment]
@tjhei, I think @gassmoeller is working through the dependent PRs (#5976, #5977). I'll update the tests after those have been merged.

**Comment by gassmoeller** [comment]
All the prerequisites have been merged. Let us know after you had the time to rebase and update the tests.

**Comment by bobmyhill** [comment]
Hi @gassmoeller , thank you! This should be ready for another round of review now.

**Pull request merged**